### ISSUE TITLE
Cursor Stuck

### ISSUE THREAD

**Issue by chrispycone** [apsun/AbsoluteTouchEx#20]
Hello
I'm having this issue with my new laptop I got and I was expecting it not gonna lie
So on my old laptop it gave 5 contact devices for the trackpad at the top when I run the program. Absolutetouch worked flawlessly and had no issues. [acer aspire v15 nitro]
>Contact for device 00020049: link=1, touchArea={0,0,1030,748}
>Contact for device 00020049: link=2, touchArea={0,0,1030,748}
>Contact for device 00020049: link=3, touchArea={0,0,1030,748}
>Contact for device 00020049: link=4, touchArea={0,0,1030,748}
>Contact for device 00020049: link=5, touchArea={0,0,1030,748}
>Detected touchpad with handle 00020049
>CreateWindowExW() -> hWnd=000308DA
>RegisterRawInputDevices(mouse)

On my new laptop [ge66 raider], it only gives 1 contact device for the trackpad.
Here is what shows up on the attest
>Contact for device 00020055: link=1, touchArea={0,0,402,244}
>Detected touchpad with handle 00020055
>CreateWindowExW() -> hWnd=000606A6
>RegisterRawInputDevices(mouse)

and as I expected when I did it with the actual program and osu! and went into absolute mode the cursor didn't move. It started clicking a bunch of things (mostly exit) or scrolling on the song selection nonstop. Is there any way I could fix this with a driver/software update? Or any other way?
If not any other way?

or am I shit outta luck with this.

**Comment by ghost** [comment]
Same problem

**Comment by AAAspen** [comment]
same problem but on an acer laptop with a precision touchpad

**Comment by now89** [comment]
yeah i dont think its going to be fixed anytime soon they stopped updating this app since 2019

**Comment by chrispycone** [comment]
Yeah it’s unfortunate so I’ll just have to learn mouse again and relative
trackpad

**Comment by now89** [comment]
i asked him on twitter waiting for a reply

**Comment by apsun** [comment]
Sorry, I was meaning to get around to this but never had the time. Truth be told I'd love to fix this, the only problem is that I don't have any devices to reproduce the issue on so any debugging attempts are just blind trial and error. I've attached a build that disables device caching which is my only guess at what the issue could be, can you run atloader.exe by itself, enable raw input mode (with shift-f6) and see if the recognized contacts change in any way?

[diag.zip](https://github.com/apsun/AbsoluteTouchEx/files/5806738/diag.zip)

If this doesn't help, I'm all out of ideas. We're just using the APIs that Windows gives us, if they don't work then either the API or the device is broken. It would be immensely helpful if you could figure out how to dump the HID descriptor of your touchpad (I tried finding a tool for this but it seems like they're all Linux-only)

**Comment by chrispycone** [comment]
> Sorry, I was meaning to get around to this but never had the time. Truth be told I'd love to fix this, the only problem is that I don't have any devices to reproduce the issue on so any debugging attempts are just blind trial and error. I've attached a build that disables device caching which is my only guess at what the issue could be, can you run atloader.exe by itself, enable raw input mode (with shift-f6) and see if the recognized contacts change in any way?
> 
> [diag.zip](https://github.com/apsun/AbsoluteTouchEx/files/5806738/diag.zip)
> 
> If this doesn't help, I'm all out of ideas. We're just using the APIs that Windows gives us, if they don't work then either the API or the device is broken. It would be immensely helpful if you could figure out how to dump the HID descriptor of your touchpad (I tried finding a tool for this but it seems like they're all Linux-only)

hey it's been a bit and I just didn't know at the time how to dump them but I managed to get them. if you're still able to help, that would be nice. If you aren't that's also fine. I did get my trackpad descriptor according to this one website.


According to https://eleccelerator.com/usbdescreqparser/ ; this is my trackpad's descriptor
04F3:30AA: Microsoft - HIDI2C Device
  05  0D  09  05  A1  01  85  04  09  22  A1  02  09  47  15  00
  25  01  35  00  45  01  65  00  55  00  75  01  95  01  81  02
  09  42  81  02  95  02  81  03  09  51  25  0F  45  00  75  04
  95  01  81  02  05  01  09  30  26  94  0C  46  92  01  65  13
  55  0E  75  10  81  02  09  31  26  A0  07  45  F4  81  02  C1
  00  05  0D  09  56  26  FF  FF  46  FF  FF  66  01  10  55  0C
  81  02  09  54  25  7F  75  08  81  02  05  09  09  01  25  01
  45  01  65  00  55  00  75  01  81  02  95  07  81  03  09  C5
  46  FF  FF  66  01  10  55  0C  75  08  95  02  81  03  85  07
  05  0D  09  60  45  01  65  00  55  00  75  01  95  01  B1  02
  96  FF  07  B1  03  85  02  09  55  25  0F  46  FF  FF  66  01
  10  55  0C  75  04  95  01  B1  02  09  59  B1  02  25  01  45
  01  65  00  55  00  75  01  96  F8  07  B1  03  85  06  06  00
  FF  09  C5  25  FF  46  FF  FF  66  01  10  55  0C  75  08  96
  00  01  B1  02  C1  00

There was also something with the same name as the trackpad, "04F3:30AA: Microsoft - HIDI2C Device" I tried figuring out what the usage was using that website, but It wouldn't give me a clear answer. So I hope this has a use to you.
  05  0D  09  0E  A1  01  85  05  09  22  A1  00  09  57  15  00
  25  01  35  00  45  01  65  00  55  00  75  01  95  01  B1  02
  09  58  B1  02  95  0E  B1  03  85  03  C1  00  09  22  A1  00
  09  52  25  0A  46  FF  FF  66  01  10  55  0C  75  10  95  01
  B1  02  C1  00  C1  00

I'm also attaching an old log I used from diag.zip
[atdebug.log](https://github.com/apsun/AbsoluteTouchEx/files/6659792/atdebug.log)

**Comment by BK8000L** [comment]
Hi, i have Acer ES1-533, ELAN touchpad.
[diag.txt](https://github.com/apsun/AbsoluteTouchEx/files/7263208/diag.txt)
![image](https://user-images.githubusercontent.com/82240117/135549039-fd9f8b11-6ed1-4732-a2c5-5200e402f1b2.png)

**Comment by BK8000L** [comment]
i have found this program, it can read my touchpad with no problems! maybe it could help
https://github.com/emoacht/RawInput.Touchpad
![image](https://user-images.githubusercontent.com/82240117/135773992-4f1c6884-4692-4dd3-ae0e-056d9e66c096.png)
1 finger means contact ID:0, 2 finger means contact ID:1...
screenshot of the creator is slightly different from mine, he has all contacts at once, I have only one contact that changes the ID.
![image](https://user-images.githubusercontent.com/82240117/135774499-d30d9467-ab1b-4d79-bf1e-60f808484eb6.png)

**Comment by chrispycone** [comment]
I doubt you still look at this dev but I used the program above me and this was the log it made.
Just by glancing at it, my trackpad doesn't react very good with multiple fingers on it causing it to spaz out between contact id 0 and whatever number of fingers.
[rawinput.txt](https://github.com/apsun/AbsoluteTouchEx/files/7289320/rawinput.txt)

**Comment by BK8000L** [comment]
>my trackpad doesn't react very good with multiple fingers on it

looks fine to me. Do you have a discord or osu account? we could talk, my discord is "0ka#5689" and osu account is "0ka"
